src/libcore/slice/mod.rs

```diff
@@ -2663,6 +2663,44 @@ impl<T> [T] {
     {
         self.iter().is_sorted_by_key(f)
     }
+
+    /// Returns index of partition point according to the given predicate,
+    /// such that all those that return true precede the index and
+    /// such that all those that return false succeed the index.
+    ///
+    /// 'self' must be partitioned.
```

Comment:
Does this mean the same as "`self` must be sorted" (for the predicate)? Can this describe ("result is unspecified") what the outcome is if this is not satisfied? Or if the predicate is not well behaved?

Reply:
The predicate returns true or false. if I can define true is smaller than false, yes self must be *sorted*.

The outcome is meaningless if not satisfied, as well as `binary_search`.

Reply:
I would word this as: The slice must be partitioned so that all elements where the predicate returns true precede the elements where the predicate returns false.

Reply:
@Amanieu Looks good. I adopted. thank you